### PR TITLE
Transaction Detail Report add mainline genres

### PR DESCRIPTION
**Pull request by taritom**
### Description:
Transaction Detail Report will now include the main and alternate genres of the headline artist.

### Migrations
 * No Migrations

### Environment Variables
 * No change

### CODE REVIEW

File: db/src/queries/reports/reports_transaction_details.sql [hunk 1]
```diff
@@ -1,76 +1,97 @@
...
+    LEFT JOIN (SELECT ti.order_item_id,
+        NULLIF(ARRAY_TO_STRING(ARRAY_AGG(DISTINCT ti.check_in_source), ', '), '') AS check_in_source
+        FROM ticket_instances ti
+        GROUP BY ti.order_item_id) AS ti_agg ON ti_agg.order_item_id = oi.id
+    LEFT JOIN holds h ON oi.hold_id = h.id
+    LEFT JOIN events e ON oi.event_id = e.id
+    LEFT JOIN users u ON COALESCE(o.on_behalf_of_user_id, o.user_id) = u.id
+    LEFT JOIN (SELECT csvg.headline_artist_alt_genres, csvg.artist_id, ea.event_id
+        FROM event_artists ea
+        INNER JOIN events e on ea.event_id = e.id
```
Comment (mikethetike) [inline db/src/queries/reports/reports_transaction_details.sql]: I don't think this join is necessary

File: db/src/queries/reports/reports_transaction_details.sql [hunk 2]
```diff
@@ -1,76 +1,97 @@
...
+        WHERE p.status IN ('Completed','Refunded')
+        GROUP BY p.order_id) AS p on o.id = p.order_id
+    LEFT JOIN (SELECT ti.order_item_id,
+        NULLIF(ARRAY_TO_STRING(ARRAY_AGG(DISTINCT ti.check_in_source), ', '), '') AS check_in_source
+        FROM ticket_instances ti
+        GROUP BY ti.order_item_id) AS ti_agg ON ti_agg.order_item_id = oi.id
+    LEFT JOIN holds h ON oi.hold_id = h.id
+    LEFT JOIN events e ON oi.event_id = e.id
+    LEFT JOIN users u ON COALESCE(o.on_behalf_of_user_id, o.user_id) = u.id
+    LEFT JOIN (SELECT csvg.headline_artist_alt_genres, csvg.artist_id, ea.event_id
```
Comment (mikethetike) [inline db/src/queries/reports/reports_transaction_details.sql]:
```suggestion
    LEFT JOIN (SELECT csvg.headline_artist_alt_genres, ea.event_id
```
You don't need to include columns that aren't used

File: db/src/queries/reports/reports_transaction_details.sql [hunk 3]
```diff
@@ -1,76 +1,97 @@
...
+        INNER JOIN (SELECT string_agg(g.name, ',') AS headline_artist_alt_genres, ag.artist_id
+		FROM artist_genres ag
+		INNER JOIN genres g ON ag.genre_id = g.id
+		GROUP BY ag.artist_id) AS csvg ON ea.artist_id = csvg.artist_id
+	    WHERE ea.rank = 0
+        GROUP BY ea.event_id, csvg.headline_artist_alt_genres, csvg.artist_id
+    ) AS csvgo ON oi.event_id = csvgo.event_id
+    LEFT JOIN (SELECT g.name AS headline_artist_main_genre, ea.event_id
+        FROM event_artists ea
+        INNER JOIN artists a ON ea.artist_id = a.id
```
Comment (mikethetike) [inline db/src/queries/reports/reports_transaction_details.sql]: I think move this table into the above `join` and `group by` it and you should only need one subquery to join to

Comment (taritom) [reply db/src/queries/reports/reports_transaction_details.sql]: Yep, thanks.

**Pull request merged**